front/src/applications/operationalStudies/views/Scenario/components/ImportTimetableItem/ImportTimetableItemTrainsList.tsx

```diff
@@ -58,16 +62,54 @@ const ImportTimetableItemTrainsList = ({
   const { t } = useTranslation('operational-studies', { keyPrefix: 'importTrains' });
 
   const {
-    train_schedules: trainSchedulesJsonData,
-    paced_trains: pacedTrainsJsonData,
+    train_schedules: trainSchedulesFromJsonData,
+    paced_trains: pacedTrainsFromJsonData,
     macro_nodes: macroNodes,
   } = trainsJsonData;
+
+  const subCategories = useSubCategoryContext();
+
+  const checkCategory = (category?: TrainCategory | null): TrainCategory | null => {
+    if (!category) return null;
+
+    // This condition is added for train imports that still use the old format: `category: string`
+    if (typeof category === 'string') {
+      if (TrainMainCategoryDict.has(category)) {
+        return { main_category: category };
+      } else {
+        return null;
+      }
```

Comment:
I think you can do something like that: 
```suggestion
const mainCategory = TrainMainCategoryDict.get(category);
return mainCategory ? { main_category: mainCategory } : null;
```

Reply:
Done